modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/application/endpoint/EndpointMatcher.java

```diff
@@ -72,10 +72,7 @@ private String _getSingleElementRegex(APIApplication.Endpoint endpoint) {
 
 		endpointPath = endpointPath + StringPool.FORWARD_SLASH;
 
-		if (Objects.equals(
-				endpoint.getPathParameter(),
-				APIApplication.Endpoint.PathParameter.ID)) {
-
+		if (Objects.equals(endpoint.getPathParameter(), "id")) {
 			return endpointPath + "\\d+";
```

Comment:
Based on official java doc, the `\\S+` means:

```
\s | A whitespace character: [ \t\n\x0B\f\r]
\S | A non-whitespace character: [^\s]
```
That means we are accepting also weird characters in the regex as a parameter. Do we want to do it? Or maybe I am missunderstanding something

Reply:
As we discussed last Friday, I'm going to accept a white space character, for `unique field` and `externalReferenceCode`. On the future we will control next rules